test/utils.spec.js

```diff
@@ -2,73 +2,88 @@
-        cb(null, val % 2 === 0)
-      }),
-      pull.collect((err, res) => {
-        expect(err).to.not.exist()
-        expect(res).to.be.eql([2, 4])
-        done()
-      })
-    )
+  it('filter - sync', async () => {
+    const data = [1, 2, 3, 4]
+    const filterer = val => val % 2 === 0
+    const res = []
+    for await (const val of utils.filter(data, filterer)) {
+      res.push(val)
+    }
+    expect(res).to.be.eql([2, 4])
   })
 
-  it('asyncFilter - async', (done) => {
-    pull(
-      pull.values([1, 2, 3, 4]),
-      utils.asyncFilter((val, cb) => {
-        setTimeout(() => {
-          cb(null, val % 2 === 0)
-        }, 10)
-      }),
-      pull.collect((err, res) => {
-        expect(err).to.not.exist()
-        expect(res).to.be.eql([2, 4])
-        done()
-      })
-    )
+  it('filter - async', async () => {
+    const data = [1, 2, 3, 4]
+    const filterer = async val => val % 2 === 0
+    const res = []
+    for await (const val of utils.filter(data, filterer)) {
+      res.push(val)
+    }
+    expect(res).to.be.eql([2, 4])
   })
 
-  it('asyncSort', (done) => {
-    pull(
-      pull.values([1, 2, 3, 4]),
-      utils.asyncSort((res, cb) => {
-        setTimeout(() => {
-          cb(null, res.reverse())
-        }, 10)
-      }),
-      pull.collect((err, res) => {
-        expect(err).to.not.exist()
-        expect(res).to.be.eql([4, 3, 2, 1])
-        done()
-      })
-    )
+  it('sortAll', async () => {
+    const data = [1, 2, 3, 4]
+    const sorter = async vals => vals.reverse()
+    const res = []
+    for await (const val of utils.sortAll(data, sorter)) {
+      res.push(val)
+    }
+    expect(res).to.be.eql([4, 3, 2, 1])
   })
 
-  it('asyncSort - fail', (done) => {
-    pull(
-      pull.values([1, 2, 3, 4]),
-      utils.asyncSort((res, cb) => {
-        setTimeout(() => {
-          cb(new Error('fail'))
-        }, 10)
-      }),
-      pull.collect((err, res) => {
-        expect(err.message).to.be.eql('fail')
-        done()
-      })
-    )
+  it('sortAll - fail', async () => {
+    const data = [1, 2, 3, 4]
+    const sorter = async vals => { throw new Error('fail') }
+    const res = []
+
+    try {
+      for await (const val of utils.sortAll(data, sorter)) {
+        res.push(val)
+      }
+    } catch (err) {
+      expect(err.message).to.be.eql('fail')
+      return
+    }
+
+    throw new Error('expected error to be thrown')
+  })
+
+  it('should take n values from iterator', async () => {
+    const data = [1, 2, 3, 4]
+    const n = 3
+    const res = []
+    for await (const val of utils.take(data, n)) {
+      res.push(val)
+    }
+    expect(res).to.be.eql([1, 2, 3])
+  })
+
+  it('should take nothing from iterator', async () => {
+    const data = [1, 2, 3, 4]
+    const n = 0
+    const res = []
+    for await (const _ of utils.take(data, n)) { // eslint-disable-line
+      throw new Error('took a value')
+    }
+    expect(res).to.be.eql([])
```

Comment:
Oh, for this test it shouldn't take any values from the iterator and so should not make it into the body of the for loop. If it does, that is an error and we throw.

Reply:
Right, but res is always `[]` as nothing is ever added to it. The expect is superfluous on a successful run, imo. You could delete `res` and the expect statement and it would be fine. The expect is really for the `for await of` to not throw if the goal is to ensure there is at least 1 expect statement. This doesn't need to hold up anything though and is fine as is.

Reply:
Gotcha, yes, you're right I'll remove!